src/bankid.ts

```diff
@@ -150,39 +151,64 @@ interface BankIdClientSettings {
   ca?: string | Buffer;
 }
 
+//
+// Error types
+//
+
+export class BankIdError extends Error {
+  readonly code: BankIdErrorCode;
+  readonly details?: string;
+
+  constructor(code: BankIdErrorCode, details?: string) {
+    super(REQUEST_FAILED_ERROR);
+    Error.captureStackTrace(this, this.constructor);
+
+    this.name = "BankIdError";
+    this.code = code;
+    this.details = details;
+  }
+}
+
+export class RequestError extends Error {
+  readonly request?: any;
+
+  constructor(request?: any) {
+    super(REQUEST_FAILED_ERROR);
+    Error.captureStackTrace(this, this.constructor);
+
+    this.name = "RequestError";
+    this.request = request;
+  }
+}
```

Comment:
Custom error classes - so now you can use `instanceof` checks for preferential error handling